### DISCUSSION TITLE
MySQL에서 Oracle로 데이터베이스 마이그레이션 진행

### DISCUSSION THREAD

**Discussion by jiwon83** [What-s-Your-Plan/wypl-server#23] [General]
MySQL에서 Oracle로 데이터베이스 마이그레이션 진행

**Comment by KIMSEI1124** [comment]
Docker로 가장 많이 사용하고 있는 오라클 이미지는 아래 사진에서 보면 [gvenzl/oracle-xe](https://hub.docker.com/r/gvenzl/oracle-xe) 입니다.

![2024-07-14_23-28-22](https://github.com/user-attachments/assets/f89cd7f1-61c1-4dc0-8938-426476c04273)

해당 이미지를 가지고 마이그레이션을 진행해볼게요.

**Comment by KIMSEI1124** [comment]
혹시 로컬환경이 아닌 원격 환경에서 해보는건 어떤가요? 맥북으로 진행할 시 설정하는게 많아서 불편함이 있습니다.

따로 데이터베이스 하나를 컨테이너로 만든 해당 데이터베이스를 혼자서 사용하는게 좋을 것 같습니다.

**Reply by jiwon83** [reply]
저는 Oracle 19버전으로 Mac 설치 [링크 ](https://www.oracle.com/kr/database/free/get-started/)참고해서 설치했습니다. 
로컬 환경에서 테스트할 때 고려해볼 수 있을 것 같습니다.

**Comment by KIMSEI1124** [comment]
@jiwon83 

저희 로컬 개발 환경에서 `H2 Database`를 사용하고 있는데 해당 데이터베이스의 모드를 `oracle`로 설정하면 오라클처럼 사용할 수 있습니다.

![2024-07-15_22-38-37](https://github.com/user-attachments/assets/b12558ea-e004-4908-8c32-59bb29a49800)

여러가지 방식이 있는데 어떤 방식이 좋을지 알려주시면 감사합니다!

[오라클 데이터베이스 설정을 추가한다. #24](https://github.com/What-s-Your-Plan/wypl-server/issues/24) 해당 이슈를 진행하고 있어서 원하시는 설정에 맞춰드릴게요.

**Reply by jiwon83** [reply]
LGTM !

**Comment by jiwon83** [comment]
[오라클 공홈](https://www.oracle.com/kr/database/free/get-started/)에서 Oracle DB 23 무료 버전을 도커 환경에서 지원된다고 합니다! 
gvenzl/oracle-xe 21 버전 이하랑 함께 고려해서 선택해보면 될 것 같습니다.
![image](https://github.com/user-attachments/assets/839523b3-ff38-4b1c-bc61-f2a0d1ab24c4)

**Reply by KIMSEI1124** [reply]
현재 오라클 21버전으로 서버를 가동하고 있습니다!

추후 23 무료 버전 도입해볼 것 이며, 일단은 마이그레이션이 끝났기 때문에 해당 게시글은 닫도록 하겠습니다!
만약 문제가 발생시 다시 열어주세요.
